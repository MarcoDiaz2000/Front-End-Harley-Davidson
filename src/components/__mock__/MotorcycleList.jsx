import React from 'react';
import PropTypes from 'prop-types';
import MotorcycleItem from './MotorcycleItem';

const MotorcycleList = ({ bikes }) => (
  <ul className="divide-customBorder">
    {bikes.map((motorcycle) => (
      <MotorcycleItem key={motorcycle.id} motorcycle={motorcycle} />
    ))}
  </ul>
);

MotorcycleList.propTypes = {
  bikes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      removed: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};
export default MotorcycleList;
