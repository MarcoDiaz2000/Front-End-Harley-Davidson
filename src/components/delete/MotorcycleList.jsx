import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchBikes } from '../../redux/bikes/thunk';
import MotorcycleItem from './MotorcycleItem';
import LoadingScreen from '../conditions/LoadingScreen';

const MotorcycleList = () => {
  const dispatch = useDispatch();
  const {
    bikes, isLoading, error, errorMsg,
  } = useSelector((state) => state.bikes);

  useEffect(() => {
    dispatch(fetchBikes());
  }, [dispatch]);

  if (isLoading) {
    return <LoadingScreen />;
  }

  if (error) {
    return (
      <div>
        <h1 className="text-white text-2xl text-center">Error</h1>
        <p className="text-white text-center">{errorMsg}</p>
      </div>
    )
  }
  
  return (
    <ul className="divide-customBorder">
      {motorcycles.length > 0 ? (
        motorcycles.map((motorcycle) => (
          <MotorcycleItem key={motorcycle.id} motorcycle={motorcycle} />
        ))
      ) : (
        <div className="text-center text-white font-bold">Without motorcycles in inventory</div>
      )}
    </ul>
  );
};

export default MotorcycleList;
