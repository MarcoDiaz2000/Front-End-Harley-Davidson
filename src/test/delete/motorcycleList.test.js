import React from 'react';
import renderer from 'react-test-renderer';
import MotorcycleList from '../../components/__mock__/MotorcycleList';

describe('MotorcycleList', () => {
  it('renders correctly', () => {
    const mockBikes = [
      { id: 1, name: 'Bike1', removed: false },
      { id: 2, name: 'Bike2', removed: true },
    ];

    const component = renderer.create(<MotorcycleList bikes={mockBikes} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
