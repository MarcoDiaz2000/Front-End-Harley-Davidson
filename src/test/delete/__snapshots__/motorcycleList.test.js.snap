// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotorcycleList renders correctly 1`] = `
<ul
  className="divide-customBorder"
>
  <li
    className="flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg bg-white white-shadow-i"
  >
    <span
      className="font-roboto font-bold text-black"
    >
      Motorcycle
    </span>
    <button
      className="p-2 rounded hover:bg-customDark-900 font-roboto font-bold text-white bg-harleyOrange"
      type="button"
    >
      Delete
    </button>
  </li>
  <li
    className="flex flex-col sm:flex-row justify-between items-center p-2 my-2 rounded-lg bg-white white-shadow-i"
  >
    <span
      className="font-roboto font-bold text-black"
    >
      Motorcycle
    </span>
    <button
      className="p-2 rounded hover:bg-customDark-900 font-roboto font-bold text-white bg-harleyOrange"
      type="button"
    >
      Delete
    </button>
  </li>
</ul>
`;
